### PR TITLE
feat: migrate rich-text-toolkit and replace usages

### PR DESCRIPTION
**Pull request by HeartSquared**
## What

Branched off #4255 

- Migrate `rich-text-toolkit`
- Remove circular dependency
- Refactor to be tree-shakeable

## Why

Migration, and remove circular dependency with RTE.

### CODE REVIEW

File: packages/components/src/RichTextEditor/EditableRichTextContent/_docs/EditableRichTextContent.mdx [hunk 1]
```diff
@@ -30,13 +30,22 @@ To render the contents of your <LinkTo pageId="components-richtexteditor">RichTe
...
+```tsx
+const [editMode, setEditMode] = useState<boolean>(false);
+const [rteData, setRTEData] = useState<EditorContentArray>([]);
 
+if (editMode) {
   return (
-    <EditableRichTextContent content={rteData} />
+    <>
+      <RichTextEditor defaultValue={rteData} />
+      <Button label="Save" primary onClick={handleCancel} />
```
Comment (gyfchong) [inline packages/components/src/RichTextEditor/EditableRichTextContent/_docs/EditableRichTextContent.mdx:41]: Save but cancel? Not sure having the button here serves much to the example though

Comment (HeartSquared) [reply packages/components/src/RichTextEditor/EditableRichTextContent/_docs/EditableRichTextContent.mdx:41]: Lol I copied it. Let me take a look and see if I can make it less confusing.

Comment (gyfchong) [reply packages/components/src/RichTextEditor/EditableRichTextContent/_docs/EditableRichTextContent.mdx:41]: I don't think its relevant to the example anyway, safe to remove 😄

**Review by gyfchong — DISMISSED** [review]

**Review by gyfchong — APPROVED** [review]

**Comment by HeartSquared** [comment]
Chromatic snapshot for Button will be fixed in a different PR. Accepted knowing it's not a change we want.

**Pull request merged**